testing-daml-models/testing-with-scenarios/5_scenario_tracing.md

```diff
@@ -0,0 +1,40 @@
+In this final step, you will learn how to add assertions and good old `printf` debugging to DAML
+scenarios.
+
+First, we'll have `Alice` check that she now really follows `Bob`:
+
+<pre class="file" data-filename="daml/User.daml" data-target="append">
+    submit alice do 
+      aliceUser <- fetch newAliceCid
+      assertMsg "Whoops!" (aliceUser.following == [bob])
+</pre>
+
+The `assertMsg` takes a text that is printed in case the assertion doesn't hold and a boolean
+expression. If you replace the last line with an assertion that doesn't hold, you'll see the
+error message in the scenario results.
+
+<pre class="file" data-filename="daml/User.daml" data-target="append">
+      assertMsg "Whoops!" $ following == [alice]
+</pre>
+
+Let's remove the failing line again.
+
+Once your scenarios grow bigger, it's helpful to trace the execution flow of your scenario. With the
+function `trace : Text -> a -> a` you can add tracing information to any expression, not just
+scenario expressions. For example, you can add a trace just before a new `User` contract is created
+after the `Follow` choice by replacing the line 
+
+<pre>
+create this with following = userToFollow :: following
+</pre>
+
+with
+
+<pre class="file" data-filename="daml/User.daml" data-target="clipboard">
+      trace "adding new follower" (create this with following = userToFollow :: following)
+</pre>
```

Comment:
If the [preview link](https://www.katacoda.com/drsk/courses/testing-daml-models/testing-with-scenarios) is the same as before then this still creates an issue with the "Copy to Editor" button

![image](https://user-images.githubusercontent.com/56260209/80693014-b1dc6e00-8ad2-11ea-8d9b-d436977d68f9.png)

Reply:
so how in the world can I simply show code?

Reply:
Dunno really, best to look at their [GitHub repo](https://github.com/katacoda/scenario-examples). Just sent an email asking them for help (you're cced)

Reply:
fixed now.